report_thesis/src/sections/proposed_approach/testing_validation.tex

```diff
@@ -47,76 +59,142 @@ \subsection{Validation and Testing Procedures for Model Evaluation}\label{subsec
 The target column $t$ refers to the column with the target variable, which would be the oxide for which we are predicting the concentration, for example, \ce{SiO_2}.
-By sorting the dataset by the target column, we ensure that the data is ordered by the target concentration values in ascending order.
+By sorting the dataset by the target column $t$, we ensure that the data is ordered by the target concentration values in ascending order.
 
 Fold numbers are then assigned sequentially using a modulo operation to ensure a random-like distribution of the unique targets across the folds (Line~\ref{line:assign_folds}).
 This means that, while the assignment process follows a sequence, the resulting distribution of targets is effectively randomized.
-If handling of extreme values is enabled, the algorithm identifies the top and bottom percentiles of the target values (Line~\ref{line:identify_extremes}) and reassigns these extreme values to the training folds (0 to \( k-2 \)) to ensure they are well-represented during model training (Line~\ref{line:reassign_extremes}).
+Fold numbers start in 0 and go up to $k-1$, as implied by the modulo operation.
+If a percentile $p$ is provided, extreme value handling is enabled, and the algorithm identifies the top and bottom percentiles of the target values (Line~\ref{line:identify_extremes}).
+These extreme values are then reassigned to the training folds (0 to $k-2$), ensuring they are evenly distributed across these folds (Line~\ref{line:reassign_extremes}).
 
-The fold assignments are then merged back into the original dataset, as described in Line~\ref{line:merge_folds}.
-Following this, the dataset is divided into a test set, which always consists of the data points assigned to fold \( k-1 \), and the remaining data forms the training set, as outlined in Line~\ref{line:split_dataset}.
-The training data is further divided into \( k-1 \) folds for cross-validation.
-For each fold, the training data consists of all but one fold, which serves as the validation set.
-These pairs of training and validation sets are then appended to the list of cross-validation folds \(\mathbf{F}_\text{cv}\) (Line~\ref{line:create_folds}).
+The fold assignments are then merged into the original dataset, as described in Line~\ref{line:merge_folds}.
+Essentially, this step enables the partitioning steps that follow, by ensuring each data item has an associated fold number.
+Following this, the dataset is divided into a test set, which always consists of the data points assigned to fold $k-1$, and the remaining data forms the training set, as outlined in Line~\ref{line:split_dataset}.
+The training data is further divided into $k-1$ sets for cross-validation. 
+For each fold $i$ where $i \in \{0, 1, \ldots, k-2\}$, we create a cross-validation training set $\mathbf{T}_\text{train}[i]$ by excluding the $i$-th fold from the set of $k-1$ folds, and use the $i$-th fold as the validation set $\mathbf{T}_\text{val}[i]$.
```

Comment:
This sentence reads a little weird. Maybe I am just not getting it, but using the $i$-th fold for the training set and the $i$-th fold as validation set, sounds to me like we are using the same fold for both.

Reply:
It says:
- create train set from all folds but $i$
- val from $i$

Need to read the `excluding` part